src/typecheck/linearization.rs

```diff
@@ -240,11 +240,10 @@ impl Completed {
 
                 self.scope_mapping
                     .get(scope)
-                    .map_or_else(|| Vec::new(), Clone::clone)
+                    .map_or_else(Vec::new, Clone::clone)
             })
             .map(|id| self.get_item(id))
-            .filter(Option::is_some)
-            .map(Option::unwrap)
+            .flatten()
```

Comment:
I don't understand why remove the `filter + map` checks since an `Option::None` value returned from `self.get_item` is still possible. + Calling `flatten` on a non-nested vector

Reply:
`Option` is an `Iterator` so the diff ends up doing the exact same thing